components/fields/number/NumberField.tsx

```diff
@@ -0,0 +1,147 @@
+import React, { ReactNode, useState } from 'react';
+import PropTypes from "prop-types";
+import { IntegerFieldModel } from "../../../types/items/IntegerFieldModel";
+import { LogService } from "../../../../core/LogService";
+import { isSafeInteger, trim } from "../../../../core/modules/lodash";
+import { FormFieldState, stringifyFormFieldState } from "../../../types/FormFieldState";
+import { ThemeService } from "../../../services/ThemeService";
+import { stringifyStyleScheme, StyleScheme } from "../../../types/StyleScheme";
+import {
+    FIELD_CLASS_NAME,
+    INTEGER_FIELD_CLASS_NAME
+} from "../../../constants/hgClassName";
+import { useNumberField } from "../../../hooks/field/useNumberField";
+import { FieldChangeCallback } from "../../../hooks/field/useFieldChangeCallback";
+import './NumberField.scss';
+
+const LOG = LogService.createLogger('IntegerField');
+const DEFAULT_PLACEHOLDER = '123.00';
+const COMPONENT_CLASS_NAME = INTEGER_FIELD_CLASS_NAME;
+const COMPONENT_INPUT_TYPE = "text";
+
+
+
+export interface NumberFieldProps {
+    readonly className   ?: string;
+    readonly style       ?: StyleScheme;
+    readonly label       ?: string;
+    readonly placeholder ?: string;
+    readonly model       ?: IntegerFieldModel;
+    readonly value       ?: number;
+    readonly change      ?: FieldChangeCallback<number | undefined>;
+    readonly changeState ?: FieldChangeCallback<FormFieldState>;
+    readonly children    ?: ReactNode;
+}
+
+ export function NumberField (props: NumberFieldProps) {
+
+    const className = props?.className;
+    const styleScheme = props?.style ?? ThemeService.getStyleScheme();
+    const placeholder = props.placeholder ?? props.model?.placeholder ?? DEFAULT_PLACEHOLDER;
+    const label = props.label ?? props.model?.label ?? '';
+    const [decimalValue, setDecimalValue] = useState<string>('');
+    const [isValid, setIsValid] = useState<boolean>(true);
+
+    const {
+        fieldState,
+        value = decimalValue,
+        onChangeCallback
+    } = useNumberField(
+        label,
+        props?.model?.key ?? '',
+        props?.change,
+        props?.changeState,
+        props?.value,
+        props?.model?.required ?? false,
+        props?.model?.minValue,
+        props?.model?.maxValue,
+        toNumber,
+        stringifyInteger
+    );
+
+    const onChangeHandler = (event:any) =>  {
```

Comment:
This should use `useCallback`. 

Also, move it as a new reusable custom hook instead under the [`hooks/fields/number`](https://github.com/heusalagroup/fi.hg.frontend/tree/main/hooks/field/number).